openlibrary/core/processors/readableurls.py

```diff
@@ -29,14 +29,14 @@ class ReadableUrlProcessor:
     The changequery function is also customized to support this.
     """
 
-    patterns = [
+    patterns: tuple[tuple[str, str, str, str], ...] = (
```

Comment:
Another case where we can drop the typehint here since it should be auto-deduced.
```suggestion
    patterns = (
```